ide-support/lsp/lsp-server/src/main/java/io/helidon/lsp/server/management/MavenSupport.java

```diff
@@ -133,7 +143,11 @@ public Set<Dependency> dependencies(final String pomPath, int timeout) {
                 return result;
             }).get(timeout, TimeUnit.MILLISECONDS);
         } catch (Exception e) {
-            LOGGER.log(Level.SEVERE, "Error when executing the maven command - " + lspMvnDependenciesCommand, e);
+            String message = "Error when executing the maven command - " + lspMvnDependenciesCommand
+                    + System.lineSeparator()
+                    + output.content().stream().collect(Collectors.joining(System.lineSeparator()));
```

Comment:
When this method fails to return the dependencies it returns `null` so it's not safe to call without checking for null. Why not return `Set.of()` instead ?

Ideally we would just throw a runtime exception as this is not recoverable.

Reply:
The value that this method returns is used later in the code and if it is equal to null the farther processing is stopped.